libvast/src/format/csv.cpp

```diff
@@ -410,16 +410,23 @@ caf::error reader::read_impl(size_t max_events, size_t max_slice_size,
                              consumer& callback) {
   VAST_ASSERT(max_events > 0);
   VAST_ASSERT(max_slice_size > 0);
+  bool timeout = false;
+  auto next_line = [&] {
+    if (!builder_ || builder_->rows() == 0)
+      lines_->next();
+    else
+      timeout = lines_->next_timeout(
+        vast::defaults::import::shared::partial_slice_read_timeout);
+  };
```

Comment:
I'd suggest to either pass in timeout as a `bool&` out-parameter or to always `return false` in the first case, passing by implicit out-reference feels a bit odd.

Reply:
I don't think this is worth refactoring, since this abstraction must necessarily change soon-ish anyways.